consensus/src/epoch_manager.rs

```diff
@@ -588,10 +601,13 @@ impl EpochManager {
             | ConsensusMsg::VoteMsg(_)
             | ConsensusMsg::CommitVoteMsg(_)
             | ConsensusMsg::CommitDecisionMsg(_) => {
-                let event: UnverifiedEvent = msg.into();
+                let event: UnverifiedEvent = msg.clone().into();
                 if event.epoch() == self.epoch() {
                     return Ok(Some(event));
-                } else {
+                } else if !matches!(
+                    msg,
+                    ConsensusMsg::CommitVoteMsg(_) | ConsensusMsg::CommitDecisionMsg(_)
```

Comment:
why? these messages carry the epoch number too

Reply:
previously we had the issue of hanging around commit phase, who keeps sending stale epoch messages, causing the node to send to itself useless epoch change proofs.

Reply:
but it shouldn't happen after we fix the commit phase?

Reply:
right, but the current commit phase fix is not clean either. We can leave it here, since it is okay to ignore commit votes and decisions with wrong epoch.

Reply:
this looks very inconsistent with other messages

Reply:
I think it has a reason as these messages are asynchronous and should not involve in the ordering?

Reply:
this is not about ordering, it's about epoch which is related to all phases.